cdap-app-fabric/src/test/java/io/cdap/cdap/internal/events/StartProgramEventSubscriberTest.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright © 2023 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.internal.events;
+
+import static org.mockito.Matchers.any;
+
+import com.google.inject.AbstractModule;
+import com.google.inject.Guice;
+import com.google.inject.Injector;
+import com.google.inject.Scopes;
+import com.google.inject.TypeLiteral;
+import io.cdap.cdap.common.app.RunIds;
+import io.cdap.cdap.common.conf.CConfiguration;
+import io.cdap.cdap.common.conf.Constants;
+import io.cdap.cdap.internal.app.services.ProgramLifecycleService;
+import io.cdap.cdap.internal.app.services.http.AppFabricTestBase;
+import io.cdap.cdap.internal.events.dummy.DummyEventReader;
+import io.cdap.cdap.internal.events.dummy.DummyEventReaderExtensionProvider;
+import io.cdap.cdap.proto.id.ProgramReference;
+import io.cdap.cdap.spi.events.EventReader;
+import io.cdap.cdap.spi.events.StartProgramEvent;
+import io.cdap.cdap.spi.events.StartProgramEventDetails;
+import java.util.ArrayList;
+import java.util.Collection;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Tests for the {@link StartProgramEventSubscriber}.
+ */
+public class StartProgramEventSubscriberTest extends AppFabricTestBase {
+  private static final Logger LOG = LoggerFactory.getLogger(StartProgramEventSubscriberTest.class);
+
+  private StartProgramEventSubscriber eventHandler;
+  private ProgramLifecycleService lifecycleService = Mockito.mock(ProgramLifecycleService.class);
+  private DummyEventReader<StartProgramEvent> eventReader = Mockito.mock(DummyEventReader.class);
```

Comment:
Suggestion: Since the DummyEventReader is intended for testing, it can take events in constructor and return that on getMessages(). Mocking will not be necessary then.